### PR TITLE
Task: Spring Boot Portfolio Project — Part 3. Executor: Oleksii Bashkirov.

### PR DESCRIPTION
**Pull request by OleksiiBashkirov**
feat: Integrate Liquibase, implement JPA repository and soft delete
- Replaced BookRepository with JpaRepository implementation.
- Added Liquibase support for database versioning.
- Changed spring.jpa.hibernate.ddl-auto property value to validate.
- Implemented soft delete functionality in the Book entity.
- Added missing endpoints and methods in BookService.
- Updated BookController to handle new endpoints:
  - GET /api/books
  - GET /api/books/{id}
  - POST /api/books
  - PUT /api/books/{id}
  - DELETE /api/books/{id}
- Updated database schema to use BIT(1) for boolean values.
- Ensured proper integration and testing with Postman.

### CODE REVIEW

File: src/main/java/mate/academy/model/Book.java [hunk 1]
```diff
@@ -10,11 +10,15 @@
...
 import org.antlr.v4.runtime.misc.NotNull;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Table(name = "books")
 @Getter
 @Setter
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Book.java]: `@Where` is deprecated, use `@SQLRestriction`

File: src/main/java/mate/academy/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
                 .toList();
     }
 
     @Override
     public BookDto findById(Long id) {
-        Book book = bookRepository.findById(id).orElseThrow(
-                        () -> new EntityNotFoundException("Book not found by id: " + id)
-                );
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/BookServiceImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
+                .orElseThrow(() -> new EntityNotFoundException("Book not found by id: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/BookServiceImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
+                .orElseThrow(() -> new EntityNotFoundException("Book not found by id: " + id));
+        book.setTitle(requestDto.getTitle());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/BookServiceImpl.java]: use MapStruct for this logic

File: src/main/java/mate/academy/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
...
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
+                .orElseThrow(() -> new EntityNotFoundException("Book not found by id: " + id));
+        book.setDeleted(true);
+        bookRepository.save(book);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/BookServiceImpl.java]:
```suggestion
        bookRepository.deleteById(id);
```

File: src/main/java/mate/academy/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
     @Override
     public List<BookDto> findAll() {
         return bookRepository.findAll().stream()
+                .filter(book -> !book.isDeleted())
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's improve your solution

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**